modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/operation/OperationMessageProcessor.java

```diff
@@ -303,7 +305,14 @@ protected ParameterValueResolver getParameterValueResolver() {
 
   @Override
   public ProcessingType getProcessingType() {
-    return asProcessingType(operationModel.getExecutionType());
+    ProcessingType processingType = asProcessingType(operationModel.getExecutionType());
+    if (processingType == CPU_LITE && !operationModel.isBlocking()) {
+      // If processing type is CPU_LITE and operation is non-blocking then use ASYNC processing type so that the Flow can return
```

Comment:
Is this something we want to do? didn't we originally intend to avoid the thread change when going back to the flow after a cpu-light process?